app/src/main/java/org/kiwix/kiwixmobile/zim_manager/local_file_transfer/SenderDeviceAsyncTask.java

```diff
@@ -0,0 +1,118 @@
+package org.kiwix.kiwixmobile.zim_manager.local_file_transfer;
+
+import android.content.ContentResolver;
+import android.net.Uri;
+import android.os.AsyncTask;
+import android.util.Log;
+import android.widget.Toast;
+
+import org.kiwix.kiwixmobile.BuildConfig;
+import org.kiwix.kiwixmobile.R;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.InetSocketAddress;
+import java.net.Socket;
+
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.FileItem.FileStatus.*;
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.WifiDirectManager.FILE_TRANSFER_PORT;
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.WifiDirectManager.copyToOutputStream;
+import static org.kiwix.kiwixmobile.zim_manager.local_file_transfer.WifiDirectManager.getFileName;
+
+/**
+ * Helper class for the local file sharing module.
+ *
+ * Once the handshake between the two connected devices has taked place, this async-task is used
+ * on the sender device to transfer the file to the receiver device at the FILE_TRANSFER_PORT port.
+ *
+ * It takes in the uri of all the files to be shared. For each file uri, creates a new connection &
+ * copies all the bytes from input stream of the file to the output stream of the receiver device.
+ * Also changes the status of the corresponding FileItem on the list of files for transfer.
+ *
+ * A single task is used by the sender for the entire transfer
+ */
+class SenderDeviceAsyncTask extends AsyncTask<Uri, Integer, Boolean> {
+
+  private static final String TAG = "SenderDeviceAsyncTask";
+
+  private WifiDirectManager wifiDirectManager;
+  private ContentResolver contentResolver;
+
+  public SenderDeviceAsyncTask(WifiDirectManager wifiDirectManager, LocalFileTransferActivity localFileTransferActivity) {
+    this.wifiDirectManager = wifiDirectManager;
+    this.contentResolver = localFileTransferActivity.getContentResolver();
+  }
+
+  @Override
+  protected Boolean doInBackground(Uri... fileUris) {
+
+    try { // Delay before trying to connect with receiver, to allow slower devices to setup server
```

Comment:
consider a method instead of a comment `delayForSlowDevicesToConnect`, stops people having to read a try catch block and get on to the meat of the method

Reply:
[Done](https://github.com/kiwix/kiwix-android/pull/1276/commits/3b1e073f0585f586a6132ac5da4591591d63eff5#diff-0431b660d7990a5ca00c1b06f9912594R50)